CDS_retriever.py

```diff
@@ -114,16 +114,28 @@ def year_retrieve(dataset, var, freq, year, grid, levelout, area, outdir, reques
             if area != 'global' :
                 retrieve_dict['area'] = area
 
-            pprint(kind)
-            pprint(level_kind)
-            pprint(retrieve_dict)
-            # run the API
+            #pprint(kind)
+            #pprint(level_kind)
+            #pprint(retrieve_dict)
+
+            # retrieve data
             c = cdsapi.Client()
-            c.retrieve(
-                kind,
-                retrieve_dict,
-                outfile)
-   
+            
+            RES = c.retrieve(
+                        kind,
+                        retrieve_dict,
+                        outfile
+                        )
+            
+            # Verify the downloaded files for completeness or errors, re-download if necessary           
+            for i in range(max_attemps):
```

Comment:
looks very good! an alternative is to try to make an error trap on the download itself, but if this works it could more than enough!